src/proto/ctl/smd.proto

```diff
@@ -72,24 +72,42 @@ message BioHealthResp {
 	uint64 cluster_size = 46; // blobstore cluster size in bytes
 }
 
-message SmdDevReq {
+enum NvmeDevState {
+	NORMAL = 0;	// Device is in a normal operational state
+	NEW = 1;	// Device is new and is not yet in-use
+	EVICTED = 2;	// Device is faulty and has been evicted
 }
 
+enum VmdLedState {
+	OFF = 0;		// Equivalent to SPDK_VMD_LED_STATE_OFF
+	QUICK_BLINK = 1;	// Equivalent to SPDK_VMD_LED_STATE_IDENTIFY	(4Hz blink)
+	ON = 2;			// Equivalent to SPDK_VMD_LED_STATE_FAULT	(solid on)
+	SLOW_BLINK = 3;		// Equivalent to SPDK_VMD_LED_STATE_REBUILD	(1Hz blink)
+	NA = 4;			// Equivalent to SPDK_VMD_LED_STATE_UNKNOWN	(VMD not enabled)
+}
+
+// SmdDevice represents a DAOS BIO device, identified by a UUID written into a label stored on a
+// SPDK blobstore created on a NVMe namespace. Multiple SmdDevices may exist per NVMe controller.
+message SmdDevice {
```

Comment:
as far as I'm aware we expect the version of dmg and daos_server to be the same as per the current interoperability guidelines, therefore this should be safe, @mjmac is that correct?

Reply:
> as far as I'm aware we expect the version of dmg and daos_server to be the same as per the current interoperability guidelines, therefore this should be safe, @mjmac is that correct?

This is correct. The interop requirements for the foreseeable future dictate that we support +1/-1 for client (including the agent)/server, but otherwise all components will be the same version. Any messages that are only used between admin/server or server/engine are fine to modify without worrying about backward compatibility. We just need to be careful about modifying messages that are used by both the agent and the server.